client/lib/rpc.rs

```diff
@@ -0,0 +1,279 @@
+
+use crate::{
+    deploy::{DeployExt, DeployParams, ListDeploys, SendDeploy, Transfer},
+    error::{Error, Result},
+};
+
+/// Struct representing a single JSON-RPC call to the casper node.
+#[derive(Debug, Default)]
+pub struct RpcCall {
+    // TODO - If/when https://github.com/AtsukiTak/warp-json-rpc/pull/1 is merged and published,
+    //        change `rpc_id` to a `jsonrpc_lite::Id`.
+    rpc_id: u32,
+    node_address: String,
+    verbose: bool,
+}
+
+/// `RpcCall` encapsulates calls made to the casper node service via JSON-RPC.
+impl RpcCall {
+    /// Creates a new RPC instance.
+    ///
+    /// `rpc_id` is used for RPC-ID as required by the JSON-RPC specification, and is returned by
+    /// the node in the corresponding response.
+    ///
+    /// `node_address` identifies the network address of the target node's HTTP server, e.g.
+    /// `"http://127.0.0.1:7777"`.
+    ///
+    /// When `verbose` is `true`, the request will be printed to `stdout`.
+    pub fn new(rpc_id: u32, node_address: String, verbose: bool) -> Self {
+        Self {
+            rpc_id,
+            node_address,
+            verbose,
+        }
+    }
+
+    /// Gets a `Deploy` from the node.
+    pub fn get_deploy(self, deploy_hash: DeployHash) -> Result<JsonRpc> {
+        let params = GetDeployParams { deploy_hash };
+        GetDeploy::request_with_map_params(self, params)
+    }
+
+    /// Queries the node for an item at `key`, given a `path` and a `state_root_hash`.
+    /// - `state_root_hash` - Hash representing a pointer to the current state from which a value
+    ///   can be extracted. See `get_state_root_hash`.
+    /// - `key` - Key from which to get a value. See (Key)[casper_types::Key].
+    /// - `path` - Path components starting with the `key`.
+    pub fn get_item(self, state_root_hash: Digest, key: Key, path: Vec<String>) -> Result<JsonRpc> {
+        let params = GetItemParams {
+            state_root_hash,
+            key: key.to_formatted_string(),
+            path,
+        };
+        GetItem::request_with_map_params(self, params)
+    }
+
+    /// Queries the node for the most recent state root hash or as of a given block hash if
+    /// `maybe_block_hash` is provided.
+    pub fn get_state_root_hash(self, maybe_block_hash: Option<BlockHash>) -> Result<JsonRpc> {
+        match maybe_block_hash {
+            Some(block_hash) => {
+                let params = GetStateRootHashParams { block_hash };
+                GetStateRootHash::request_with_map_params(self, params)
+            }
+            None => GetStateRootHash::request(self),
+        }
+    }
+
+    /// Gets the balance from a purse.
+    /// - `state_root_hash` - Hash representing a pointer to the current state from which a balance
+    ///   can be extracted. See `get_state_root_hash`.
+    /// - `purse_uref` - Represents the purse from which the balance will be obtained. Can be
+    ///   obtained with `get_item`.
```

Comment:
I think it might be clearer to specify that it's the URef which can be obtained with `get_item`?